DSCResources/MSFT_xIisFeatureDelegation/MSFT_xIisFeatureDelegation.psm1

```diff
@@ -0,0 +1,160 @@
+ChangedMessage=Changed overrideMode for '{0}' to {1}
+'@
+}
+
+######################################################################################
+# The Get-TargetResource cmdlet.
+# This function will get the Mime type for a file extension
+######################################################################################
+function Get-TargetResource
+{
+  [OutputType([Hashtable])]
+	param
+	(		
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+		[String]$SectionName,
+
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+        [ValidateSet("Allow", "Deny")]
+        [String]$OverrideMode
+	)
+	
+    CheckIISPoshModule
+
+    $getTargetResourceResult = $null;
+
+    [string]$oMode = GetOverrideMode $SectionName
+
+    if ($oMode -eq $null)
+    {
+        $ensureResult = "Absent";
+    }
+    else
+    {
+        $ensureResult = "Present"
+        $getTargetResourceResult = @{SectionName = $SectionName
+                                    OverrideMode = $oMode}
+    }
+
+    
+	return $getTargetResourceResult
+}
+
+######################################################################################
+# The Set-TargetResource cmdlet.
+# This function set the OverrideMode for a given section if not already correct
+######################################################################################
+function Set-TargetResource
+{
+	param
+	(	
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+		[String]$SectionName,
+
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+        [ValidateSet("Allow", "Deny")]
+        [String]$OverrideMode
+    )
+
+    CheckIISPoshModule
+    [string]$oMode = GetOverrideMode $SectionName
+
+
+    if ($oMode -eq "Allow" -and $OverrideMode -eq "Deny")
+    {
+         Set-webconfiguration -Location "" -Filter "/system.webServer/$SectionName" -PSPath "machine/webroot/apphost" -metadata overrideMode -value Deny
+         Write-Verbose($($LocalizedData.ChangedMessage) -f $SectionName,"Deny")
+    }
+    elseif ($oMode -eq "Deny" -and $OverrideMode -eq "Allow")
+    {
+         Set-webconfiguration -Location "" -Filter "/system.webServer/$SectionName" -PSPath "machine/webroot/apphost" -metadata overrideMode -value Allow
+         Write-Verbose($($LocalizedData.ChangedMessage) -f $SectionName,"Allow")
+    }
+}
+
+######################################################################################
+# The Test-TargetResource cmdlet.
+# This will test if the given section has the required OverrideMode
+######################################################################################
+function Test-TargetResource
+{
+    [OutputType([System.Boolean])]
+	param
+	(	
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+		[String]$SectionName,
+
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+        [ValidateSet("Allow", "Deny")]
+        [String]$OverrideMode
+	)
+
+    CheckIISPoshModule
+
+    [bool]$DesiredConfigurationMatch = $true;
```

Comment:
It's not necessary to cast this to a bool as the assignment will do that for you.